File: _assets/output_gen/jsModels/Registrant.js

//---------------------------------------------------------------------------

export default class Registrant {
  constructor(options) {

    this.Id = null;
    this.EncKey = null;
    this.Code = null;
    this.FirstName = null;
    this.MiddleName = null;
    this.LastName = null;
    this.Suffix = null;
    this.DOB = null;
    this.SSN = null;
    this.StateIdNumber = null;
    this.Street1 = null;
    this.Street2 = null;
    this.City = null;
    this.StateCode = null;
    this.PostalCode = null;
    this.DateCreated = null;
    this.DateUpdated = null;
    this.DateVerified = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.EncKey = options.EncKey || this.EncKey;
    this.Code = options.Code || this.Code;
    this.FirstName = options.FirstName || this.FirstName;
    this.MiddleName = options.MiddleName || this.MiddleName;
    this.LastName = options.LastName || this.LastName;
    this.Suffix = options.Suffix || this.Suffix;
    this.DOB = options.DOB || this.DOB;
    this.SSN = options.SSN || this.SSN;
    this.StateIdNumber = options.StateIdNumber || this.StateIdNumber;
    this.Street1 = options.Street1 || this.Street1;
    this.Street2 = options.Street2 || this.Street2;
    this.City = options.City || this.City;
    this.StateCode = options.StateCode || this.StateCode;
    this.PostalCode = options.PostalCode || this.PostalCode;
    this.DateCreated = options.DateCreated || this.DateCreated;
    this.DateUpdated = options.DateUpdated || this.DateUpdated;
    this.DateVerified = options.DateVerified || this.DateVerified;

  }  
  //---------------------------------------------------------------------------  
}

File: src/OB.Data/Entities/Registrant.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Registrant
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string EncKey { get; set; }                     // private "salt" for encrypting the signals

        [Required, StringLength(24)]
        public string Code { get; set; }                       // private identifier


        //  demographic

        [Required, StringLength(32)]
        public string FirstName { get; set; }

        [Required, StringLength(32)]
        public string LastName { get; set; }

        [StringLength(32)]
        public string MiddleName { get; set; }

        [StringLength(32)]
        public string Suffix { get; set; }


        //  validation

        [Required]
        public DateTime DOB { get; set; }

        [StringLength(16)]
        public string SSN { get; set; }

        [Required, StringLength(16)]
        public string StateIdNumber { get; set; }


        //  residency

        [Required, StringLength(32)]
        public string Street1 { get; set; }

        [StringLength(32)]
        public string Street2 { get; set; }

        [Required, StringLength(32)]
        public string City { get; set; }

        [Required, StringLength(2)]
        public string StateCode { get; set; }

        [Required, StringLength(10)]
        public string PostalCode { get; set; }

        // ----------------------------------------------  audit
        [Required]
        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }

        public DateTime? DateVerified { get; set; }
    }
}
